src/helpers.js

```diff
@@ -56,40 +58,75 @@ export const setupCartUpdateListener = callback => {
   });
 };
 
+export const getConstants = async () => {
+  try {
+    const docRef = await getConstantsQuery.get();
+    const respJson = await (docRef.exists && docRef.data());
+    return respJson;
+  } catch (error) {
+    console.error("Encountered error getting constants: ", error.message);
+  }
+};
+
+export const setupTriggersUpdateListener = async callback => {
+  getTriggersQuery.onSnapshot(async docRef => {
+    try {
+      const respJson = await (docRef.exists && docRef.data());
+      callback(respJson);
```

Comment:
callbacks inside async/await :0

Reply:
Idk how else you would do that? Because you don't wanna just return, you want the `callback` to be able to do something with the data every time